webapp/javascript/services/users.ts

```diff
@@ -81,6 +81,45 @@ export async function loadCurrentUser(): Promise<
   return Result.err<User, RequestError>(response.error);
 }
 
+export async function logIn({
+  username,
+  password,
+}: {
+  username: string;
+  password: string;
+}): Promise<any> {
+  const response = await request(`/login`, {
+    method: 'POST',
+    body: JSON.stringify({ username, password: passwordEncode(password) }),
+  });
+  if (response.isOk) {
+    return Promise.resolve(response.value);
+  }
+
+  return Promise.reject(response.error);
```

Comment:
The pattern we are following is to never fail at the service level, and let that be handled at the component level. Can you return a `Result/Err` instead?